components/hab/src/command/pkg/binlink.rs

```diff
@@ -178,20 +193,57 @@ struct Binlink {
 
 #[cfg(not(target_os = "windows"))]
 impl Binlink {
-    pub fn new<T: AsRef<Path>>(src: T, dest_dir: T) -> Result<Self> {
+    pub fn new<T: AsRef<Path>>(
+        src: T,
+        dest_dir: T,
+        stub: T,
+        pkg_ident: &PackageIdent,
+    ) -> Result<Self> {
+        use std::os::unix::fs::OpenOptionsExt;
+
         let bin_name = match src.as_ref().file_name() {
             Some(name) => name,
             None => return Err(Error::CannotParseBinlinkSource(src.as_ref().to_path_buf())),
         };
 
-        Ok(Self {
-            dest: dest_dir.as_ref().join(bin_name),
-            src: src.as_ref().to_path_buf(),
-        })
+        let template = format!(
+            "\
+#!/bin/sh 
+if ! [ -x \"$(command -v hab)\" ]; then 
+    echo 'ERROR: The core/hab package needs to be binlinked' 
+    echo
+    exit 1 
+fi 
+hab pkg exec {0} {1} \"$@\"",
+            pkg_ident,
+            bin_name.to_string_lossy()
+        );
+
+        // Need to treat core/hab differently otherwise we get into a link loop
+        match format!("{}/{}", pkg_ident.origin, pkg_ident.name).as_ref() {
```

Comment:
You shouldn't need to allocate a new string to check if this is `core/hab`... I'd just check if the origin was "core" and the name was "hab", and go from there.

Reply:
Ha, fixed. No idea why I chose this way of doing it. ☕️ needed maybe,